crates/edr_provider/src/data.rs

```diff
@@ -146,6 +157,17 @@ impl ProviderData {
         })
     }
 
+    pub async fn reset(&mut self, fork_config: Option<ForkConfig>) -> Result<(), CreationError> {
+        let mut config = self.initial_config.clone();
+        config.fork = fork_config;
+
+        let mut reseted = Self::new(self.runtime_handle.clone(), config).await?;
+
+        std::mem::swap(self, &mut reseted);
```

Comment:
```suggestion
        let mut reset = Self::new(self.runtime_handle.clone(), config).await?;

        std::mem::swap(self, &mut reset);
```

Grammar

Reply:
I called this `reseted` even though I know it's grammatically incorrect, because I found it difficult to read when using `reset` in the imperative in the method name and then as past participle in the method body. I renamed it `reset_instance` now in [e396b25](https://github.com/NomicFoundation/hardhat/pull/4665/commits/e396b252b43322252c6fcf8cee95179410041649) which is both grammatically correct and easier to read imo.